### ISSUE TITLE
Should warn if for loop clobbers array variable index 0

### ISSUE THREAD

**Issue by zhatt** [koalaman/shellcheck#2730]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

options=( a b c )

for outer in 1 2; do
	# Accidentally clobbering options[0] meant to use option
	for options in "${options[@]}"; do
		echo "$outer:${options[0]},${options[1]},${options[2]}"
	done
done

```

#### Here's what shellcheck currently says:
```

```
#### Here's what I wanted or expected to see:

TODO: for loop variable should not be the same as array variable that is being looped over.
